### ISSUE TITLE
I keep seeing  this: TypeError: external_root_React_commonjs2_react_commonjs_react_amd_react_default.a is undefined

### ISSUE THREAD

**Issue by talkingtab** [react-bootstrap/react-bootstrap#5086]
I"m trying to  use the Accorion component with Parcel. This works in the development environment but fails in production. Don't know where to start.

**Comment by mxschmitt** [comment]
Hey, I've added two Parcel related examples to our CodeSandbox getting started. One with JavaScript and one with TypeScript, see this for a reference: https://github.com/react-bootstrap/code-sandbox-examples/pull/74

**Comment by talkingtab** [comment]
Hey Max, thank you. I tried to open the examples on CodeSandBox and could not get access. I'm not why - I'm not an experienced sandbox user. I'm stuck on this, so if it is possible to see the example another way I would appreciate it. Best, Terry

**Comment by mxschmitt** [comment]
> Hey Max, thank you. I tried to open the examples on CodeSandBox and could not get access. I'm not why - I'm not an experienced sandbox user. I'm stuck on this, so if it is possible to see the example another way I would appreciate it. Best, Terry

Yep there are not merged yet, thats the reason. I've linked it for code reference, so you can see how I did integrate it with Parcel.

Will be merged soon, then I'll send the interactive URL over.

**Comment by talkingtab** [comment]
Thanks. 
Steps to reproduce: 
1. copy the parcel example to accordion (cp -r parcel accordion)
2. Edit package.json in accordion for parcel build so that you can open the build as a file
```
 "build": "parcel build src/index.html --public-url ."
```
3. Edit src/index.js to have the following code from the Accordion Component Example
```
import React, { useState } from "react";
import ReactDOM from "react-dom";

// Importing the Bootstrap CSS
import "bootstrap/dist/css/bootstrap.min.css";

import Jumbotron from "react-bootstrap/Jumbotron";
import Container from "react-bootstrap/Container";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import Accordion from "react-bootstrap/Accordion";

const App = () => ( 
    <Container className="p-3">
        <Jumbotron>
            <h1 className="header">Welcome To React-Bootstrap</h1>
            <Accordion defaultActiveKey="0">
                <Card>
                    <Card.Header>
                        <Accordion.Toggle as={Button} variant="link" eventKey="0">
                            Click me!
                        </Accordion.Toggle>
                    </Card.Header>
                    <Accordion.Collapse eventKey="0">
                        <Card.Body>Hello! I'm the body</Card.Body>
                    </Accordion.Collapse>
                </Card>
                <Card>
                    <Card.Header>
                        <Accordion.Toggle as={Button} variant="link" eventKey="1">
                            Click me!
                        </Accordion.Toggle>
                    </Card.Header>
                    <Accordion.Collapse eventKey="1">
                        <Card.Body>Hello! I'm another body</Card.Body>
                    </Accordion.Collapse>
                </Card>
            </Accordion>
        </Jumbotron>
    </Container>
);

var mountNode = document.getElementById("root");
ReactDOM.render(<App />, mountNode);

```
4. Open the file `open dist/index.html`

The modal is not open, and clicking on the "click me" buttons does nothing.

5. Run `yarn run start` and open `localhost:1234` and everything works as expected.

**Comment by mxschmitt** [comment]
Thank for for the example, it was really helpful. But unfortunately it semems like this is a upstream bug in Parcel: https://github.com/parcel-bundler/parcel/issues/3523

**Comment by talkingtab** [comment]
Thanks! I got this to work by using parcel@next. I appreciate the help!

```
targets: { 
     browser: {
        publicUrl: "./"
    }
}
```

**Comment by talkingtab** [comment]
It appears that parcel@next is not far enough along to be feasible now. I'm wondering if someone could give me a one liner about this problem. It *appears* that the issue is simply that AccordianContext.js is a two line file that is "optimized" by  parcel because it is **exactly the same** CardContext.js and probably other files. 

Is it possible to for me to clone the project and add some way distinguishing feature to the Context? I can play around with it, but would welcome some thoughts, insights. Thanks.

**Comment by kosmiq** [comment]
Reading the issue here: https://github.com/parcel-bundler/parcel/issues/3523

> With parcel, these "two modules" actually execute only once and return the same value. If I add some comment to one (e.g. //) it works as expected.

So if you fork and add comments within either one you should be fine for now.

**Comment by talkingtab** [comment]
Thanks, I missed that. After years of Create-React-App experience, I think parcel is well worth using. It would be nice to add something to the two Parcel examples that notes this problem?

**Comment by jquense** [comment]
Yes lets just implement this here and avoid ppl having to fork

**Comment by talkingtab** [comment]
@jquense 
Yes Please! I did the fork, but trying to do a build so I can npm install was too much. Right now every time I do a new install, I manually edit node_modules/react-bootstrap/*/NavbarContext, CardContext, and AccordionContext and add a line 
```
var name = "card" 
``` 
or whatever.

**Comment by kyletsang** [comment]
This is fixed in #5181

**Comment by CGenie** [comment]
I just installed `react-bootstrap` via yarn, version 1.5.2 and I'm still seeing this error. Line 1425 of the file `react-bootstrap.js` looks like this:
```js
var ThemeContext = /*#__PURE__*/external_root_React_commonjs2_react_commonjs_react_amd_react_default().createContext({});
```
and throws an error.

**Comment by kyletsang** [comment]
@CGenie can you provide a minimal reproduction?  This issue is very old now, so it might be caused by something else.

**Comment by CGenie** [comment]
Thing is, it's hard to reproduce. I just installed this via `yarn add`.

**Comment by talkingtab** [comment]
The original problem I experienced had to do with Parcel. Parcel finds duplicates of  files and removes them. So for example a package contains two files each with the same name and each that says "const name= 'dog'", then Parcel removes one of the files. I was importing two components each of which had the same file with the exact same name and exact same parsed contents. You could work around the problem by doing a postinstall change, adding an innocuous distinguisher to the files. For example "const component = 'accordion'" to one and "const component = 'menu'" to the other. I have no idea if this is still the case.

**Comment by CGenie** [comment]
@talkingtab Do you think this is the issue here? How do people work with this code?
I basically just `yarn add`, then I add the `node_modules/react-bootstrap/dist/react-bootstrap.js` file to my scripts in `index.html`.

**Comment by CGenie** [comment]
In fact if I just include the script from this page
https://react-bootstrap.github.io/getting-started/introduction/
like this:
```html
<script
  src="https://unpkg.com/react-bootstrap@next/dist/react-bootstrap.min.js"
  crossorigin></script>
```
then I get this error:
```
Uncaught TypeError: r is undefined
    356 http://localhost:8008/js/react-bootstrap.unpkg.min.js:2
```

**Comment by CGenie** [comment]
Ah OK but with fresh `index.html` file with only the contents from the docs, it seems to work. So my guess it's some libs missing on my side...
